### DISCUSSION TITLE
error access to public variable

### DISCUSSION THREAD

**Discussion by zach030** [foundry-rs/foundry#5333] [Help]
Hello, I am writing testFunction for contract but encounter compile error. 
For example I have contract A like this:
```solidity
contract A {
    uint256 public val;
}
```
And in my test function, when I write the code like `a.val()` to access to public variable defined in contract A then I got the compile error: `No matching declaration found after argument-dependent lookup.solidity(9322)`

I was wondering how to access to public variable in founrdry test function? Thanks!

**Comment by allwin199** [comment]
can you also show your test fn
